src/test/java/com/nylas/ContactTest.java

```diff
@@ -0,0 +1,165 @@
+
+        List<Contact.PhysicalAddress> physical_addresses = new LinkedList<>();
+        Contact.PhysicalAddress physicalAddress = new Contact.PhysicalAddress();
+        physicalAddress.setFormat("standard");
+        physicalAddress.setStreetAddress("1234");
+        physicalAddress.setCity("Portland");
+        physicalAddress.setCountry("U.S");
+        physicalAddress.setState("Oregon");
+        physicalAddress.setType("physical");
+        physicalAddress.setPostalCode("0101");
+        physical_addresses.add(physicalAddress);
+        contact.setPhysicalAddresses(physical_addresses);
+
+        List<Contact.PhoneNumber> phone_numbers = new LinkedList<>();
+        Contact.PhoneNumber phoneNumber = new Contact.PhoneNumber("cellphone", "091283");
+        phone_numbers.add(phoneNumber);
+        contact.setPhoneNumbers(phone_numbers);
+
+
+        List<Contact.WebPage> web_pages = new LinkedList<>();
+        Contact.WebPage webPage = new Contact.WebPage("page", "linux-foundation.org");
+        web_pages.add(webPage);
+        contact.setWebPages(web_pages);
+
+        ContactGroup contactGroup = new ContactGroup();
+        setField("name", "org", contactGroup);
+        setField("path", "it", contactGroup);
+        contact.setGroup(contactGroup);
+
+        setField("source", "address_book", contact);
+    }
+
+    @Test
+    public void testGetters() {
+        assertEquals(contact.getObjectType(), "contact");
+        assertEquals(contact.getGivenName(), "Linus");
+        assertEquals(contact.getMiddleName(), "Benedict");
+        assertEquals(contact.getSurname(), "Torvalds");
+        assertEquals(contact.getSuffix(), "mr");
+        assertEquals(contact.getNickname(), "Linux");
+        assertEquals(contact.getBirthday(), "December 28, 1969");
+        assertEquals(contact.getCompanyName(), "Linux Foundation");
+        assertEquals(contact.getJobTitle(), "Founder");
+        assertEquals(contact.getManagerName(), "N/A");
+        assertEquals(contact.getOfficeLocation(), "Portland, Oregon");
+        assertEquals(contact.getNotes(), "Known for git and linux");
+        assertEquals(contact.getPictureUrl(), "https://media.newyorker.com/photos/5ba177da9eb2f7420aadeb98/master/w_1920,c_limit/Cohen-Linus-Torvalds.jpg");
+        assertEquals(contact.getEmails().size(), 1);
+        assertEquals(contact.getIMAddresses().size(), 1);
+        assertEquals(contact.getPhysicalAddresses().size(), 1);
+        assertEquals(contact.getPhoneNumbers().size(), 1);
+        assertEquals(contact.getWebPages().size(), 1);
+        assertEquals(contact.getGroups().size(),1);
+        assertEquals(contact.getSource(), "address_book");
+    }
+
+    @Test
+    public void testGetWriteableFields_creation() {
+        assertEquals(contact.getWritableFields(true).size(), 16);
+    }
+
+    @Test
+    public void testGetWriteableFields_NoCreation() {
+        assertEquals(contact.getWritableFields(false).size(), 16);
+    }
```

Comment:
I know it seems a little redundant, but can you also test for the specific fields in the list? Like

```java
writableFields = contact.getWritableFields(true);
assertEquals(writableFields.get("notes"), "Known for git and linux");
// etc.
```

It will be handy for other classes as well moving forward since some of the writable fields change if creation == true or false!

Reply:
Hi @mrashed-dev I added the scenarios that you mentioned.
It was worth it as it describes the behavior of the `group` field.